sensors/triggers/slack/slack.go

```diff
@@ -114,54 +115,65 @@ func (t *SlackTrigger) Execute(ctx context.Context, events map[string]*v1alpha1.
 
 	api := slack.New(slackToken, slack.OptionDebug(false))
 
-	params := &slack.GetConversationsParameters{
-		Limit: 200,
-		Types: []string{"public_channel", "private_channel"},
-	}
-	channelID := ""
-	isPrivateChannel := false
+	t.Logger.Infow("posting to channel...", zap.Any("channelName", channel))
 	for {
-		channels, nextCursor, err := api.GetConversations(params)
+		channelID, timestamp, err := api.PostMessage(channel, slack.MsgOptionText(message, false))
```

Comment:
With the code change, do we still need the for loop in line 119? I seems to be infinite.

Reply:
This is to implement retry logic, i.e. try to post message first, then if it fails due to not_in_channel error, join the channel and try again.

It shouldn't be infinite, as all paths of logic inside the loop lead to a return, either due to success or error. If the `api.JoinConversation` call on line 161 incorrectly returns a `nil` value for `err` when it actually did fail under the hood, this could loop infinitely. Not sure how likely that is, but I could limit the loop to max 2 iterations and add some sort of "unspecified error" logging after it.

Reply:
I see.